### PR TITLE
[NOT FOR MERGE] Common base rebased

### PR DESCRIPTION
**Pull request by mkopec**
Rebased on top of latest master. Most conflicts were related to the TianoCore payload being renamed to edk2, and some of our config options became redundant. Squashed all missing edk2 changes into a single commit.

Supersedes https://github.com/Dasharo/coreboot/pull/201

### CODE REVIEW

File: util/genbuild_h/genbuild_h.sh [hunk 1]
```diff
@@ -66,6 +66,10 @@ printf "#define COREBOOT_VERSION %s\n" "\"$KERNELVERSION\""
 
 #See if the build is running in a git repo and the git command is available
 printf "/* timesource: $TIMESOURCE */\n"
+
+printf "#define DASHARO_VERSION \"%s\"\n" `git describe --tags --match "asus_kgpe-d16_v[0-9]*.[0-9]*.[0-9]*" | sed 's/asus\_kgpe\-d16\_//' | sed 's/\(v[0-9]\.[0-9]\.[0-9]\+\).*/\1/'`
+printf "#define DASHARO_MAJOR_VERSION %d\n#define DASHARO_MINOR_VERSION %d\n#define DASHARO_PATCH_VERSION %d\n" `git describe --tags --match "asus_kgpe-d16_v[0-9]*.[0-9]*.[0-9]*" | sed 's/asus\_kgpe\-d16\_//' | sed 's/v\([0-9]\)\.\([0-9]\)\.\([0-9]\+\).*/\1 \2 \3/'`
```
Comment (pietrushnic) [inline util/genbuild_h/genbuild_h.sh:71]: I assume this is still problematic, it should not be paltform specific?

**Comment by pietrushnic** [comment]
@mkopec I would be glad to test it e.g. with Dell OptiPlex, also I assume my branch with optiplex support should be rebased on this?

**Comment by mkopec** [comment]
> @mkopec I would be glad to test it e.g. with Dell OptiPlex, also I assume my branch with optiplex support should be rebased on this?

Probably it should be rebased, yes

File: util/genbuild_h/genbuild_h.sh (hunk 1 above)
Comment (mkopec) [reply util/genbuild_h/genbuild_h.sh:71]: Right, it's still a problem

**Comment by mkopec** [comment]
Added the SecureBoot default state Kconfig option.

File: payloads/external/edk2/Kconfig [hunk 2]
```diff
@@ -188,6 +188,29 @@ config EDK2_SERIAL_SUPPORT
 	  Enable serial port output in edk2. Serial output limits the performance of edk2's
 	  FrontPage.
 
+config EDK2_ENABLE_IPXE
```
Comment (mkopec) [inline payloads/external/edk2/Kconfig:191]: Should also add custom boot menu and setup menu keys

File: util/genbuild_h/genbuild_h.sh (hunk 1 above)
Comment (mkopec) [reply util/genbuild_h/genbuild_h.sh]: Changed to get DASHARO_VERSION and others from CONFIG_LOCALVERSION. If that's not set, DASHARO_VERSION will be set to `*INVALID*`

File: payloads/external/edk2/Kconfig (hunk 2 above)
Comment (mkopec) [reply payloads/external/edk2/Kconfig:191]: Done

**Comment by mkopec** [comment]
Removed duplicated submodule urls and squashed everything down to 8 commits

**Comment by mkopec** [comment]
Could also include persistent bootlogo functionality in here.

File: payloads/external/edk2/Makefile [hunk 3]
```diff
@@ -32,19 +32,19 @@ endif
 ifeq ($(CONFIG_EDK2_RELEASE),y)
 BUILD_STR += -b RELEASE
 endif
-# DISABLE_SERIAL_TERMINAL	= FALSE
-ifneq ($(CONFIG_EDK2_SERIAL_SUPPORT),y)
-BUILD_STR += -D DISABLE_SERIAL_TERMINAL=TRUE
+# SERIAL_TERMINAL	= TRUE
+ifeq ($(CONFIG_EDK2_SERIAL_SUPPORT),y)
+BUILD_STR += -D SERIAL_TERMINAL=TRUE
```
Comment (mkopec) [inline payloads/external/edk2/Makefile:37]: Could also flip the logic on EDK2's side, as upstream does it.

File: src/arch/x86/smbios.c [hunk 4]
```diff
@@ -277,76 +277,38 @@ static int create_smbios_type17_for_dimm(struct dimm_info *dimm,
...
 						    sizeof(*t), handle);
 
-	t->vendor = smbios_add_string(t->eos, "coreboot");
+	char bversion[100];
+
+	t->vendor = smbios_add_string(t->eos, "3mdeb");
 	t->bios_release_date = smbios_add_string(t->eos, coreboot_dmi_date);
 
+	snprintf(bversion, sizeof(bversion), "Dasharo (%s) %s",
+		get_bios_framework_string(), dasharo_version);
```
Comment (miczyg1) [inline src/arch/x86/smbios.c:302]:
```suggestion
		get_bios_framework_string(), CONFIG_LOCALVERSION);
```

Not sure if you noticed, but all those dasharo_version and dasharo_{minor, major, patch}_revision fields are nowhere used. If we are to sue the LOCALVERSION anyways we can simply use it directly.

Comment (mkopec) [reply src/arch/x86/smbios.c:302]: Ah, the major and minor versions were previously used in `system_bios_{major,minor)_release`, but that apparently that was lost, will fix

File: payloads/external/edk2/Makefile [hunk 5]
```diff
@@ -76,6 +76,34 @@ endif
...
+BUILD_STR += -D OPAL_PASSWORD_ENABLE=TRUE
+endif
+# TPM_ENABLE		= TRUE
+ifeq ($(CONFIG_EDK2_TPM_ENABLE),y)
+BUILD_STR += -D TPM_ENABLE=TRUE
+endif
+
+BUILD_STR += -D BOOT_MENU_KEY=$(CONFIG_EDK2_BOOT_MENU_KEY)
+BUILD_STR += -D SETUP_MENU_KEY=$(CONFIG_EDK2_SETUP_MENU_KEY)
+BUILD_STR += -D PLATFORM_BOOT_TIMEOUT=$(CONFIG_EDK2_BOOT_TIMEOUT)
```
[the remark this answers is not in the document]
Comment (mkopec) [reply payloads/external/edk2/Makefile]: Fixed

File: src/arch/x86/smbios.c (hunk 4 above)
Comment (mkopec) [reply src/arch/x86/smbios.c:302]: Added. `dasharo_patch_revision` remains unused

**Comment by pietrushnic** [comment]
Ok, I will rebase OptiPlex code on top of this series. More to that I would like to use our laptop automated testing to check quality in context of SMBOIS tables. I should have something for OptiPlex Tuesday.

**Comment by pietrushnic** [comment]
@miczyg1 @macpijan our goal here should be to upstream all above changes, all at least decouple those from implementation. For example most of the stuff should be determined by config instead of hardcoded values. In case of modules, maybe there is a way to not have relative paths in upstream or change repo initial cloning method so it can have correct relation with submodules.

**Comment by pietrushnic** [comment]
@mkopec @miczyg1 I rebased on this branch and removed all commits related to tianocore since it was renamed to but I'm getting compilation error:

```
    Fetching new commits from https://github.com/Dasharo/edk2.git
    Checking out edk2 revision origin/dasharo
HEAD is now at 17b0d5b1b2 add debug prints that made optiplex booting
EDK2: Checking uuid-dev:    Found!
EDK2: Checking nasm:        Found!
cp: cannot create regular file '/MdeModulePkg/Logo/Logo.bmp': No such file or directory
make[1]: *** [Makefile:155: logo] Error 1
make: *** [payloads/external/Makefile.inc:158: build/UEFIPAYLOAD.fd] Error 2
```

I have 3rdpary/dasharo with bmp, but it see like `project_dir` is not defined in edk2 payload Makefile.

**Comment by pietrushnic** [comment]
@miczyg1 how to introduce that? Rebase again?

File: payloads/external/edk2/Makefile [hunk 6]
```diff
@@ -127,10 +154,10 @@ $(EDK2_PATH): $(WORKSPACE)
...
 	case "$(CONFIG_EDK2_BOOTSPLASH_FILE)" in \
 		"") ;; \
-		/*) convert -background None $(CONFIG_EDK2_BOOTSPLASH_FILE) \
-			BMP3:$(EDK2_PATH)/MdeModulePkg/Logo/Logo.bmp;; \
-		*) convert -background None $(top)/$(CONFIG_EDK2_BOOTSPLASH_FILE) \
-			BMP3:$(EDK2_PATH)/MdeModulePkg/Logo/Logo.bmp;; \
+		/*) cp $(CONFIG_EDK2_BOOTSPLASH_FILE) \
+				$(project_dir)/MdeModulePkg/Logo/Logo.bmp;; \
+		*) cp $(top)/$(CONFIG_EDK2_BOOTSPLASH_FILE) \
+				$(project_dir)/MdeModulePkg/Logo/Logo.bmp;; \
```
Comment (miczyg1) [inline payloads/external/edk2/Makefile:160]:
```suggestion
		/*) cp $(CONFIG_EDK2_BOOTSPLASH_FILE) \
				$(EDK2_PATH)/MdeModulePkg/Logo/Logo.bmp;; \
		*) cp $(top)/$(CONFIG_EDK2_BOOTSPLASH_FILE) \
				$(EDK2_PATH)/MdeModulePkg/Logo/Logo.bmp;; \
```

**Comment by miczyg1** [comment]
> @miczyg1 how to introduce that? Rebase again?

EDK2_PATH is already exported in the Makefile, added suggestion to fix the problem.

**Comment by pietrushnic** [comment]
```
ASSERT_EFI_ERROR (Status = Unsupported)
ASSERT [UiApp] /home/coreboot/coreboot/payloads/external/edk2/workspace/Build/UefiPayloadPkgX64/DEBUG_COREBOOT/X64/MdeModulePkg/Application/UiApp/UiApp/DEBUG/AutoGen.c(715): !EFI_ERROR (Sta)
```

I saw that one before. What we can do with it?

**Comment by miczyg1** [comment]
> ```
> ASSERT_EFI_ERROR (Status = Unsupported)
> ASSERT [UiApp] /home/coreboot/coreboot/payloads/external/edk2/workspace/Build/UefiPayloadPkgX64/DEBUG_COREBOOT/X64/MdeModulePkg/Application/UiApp/UiApp/DEBUG/AutoGen.c(715): !EFI_ERROR (Sta)
> ```
> 
> I saw that one before. What we can do with it?

Yeah, unless you use debug build you won't catch that... This is the lien that needs change:
https://github.com/Dasharo/DasharoModulePkg/blob/main/Library/DasharoSystemFeaturesUiLib/DasharoSystemFeatures.c#L72

It should return EFI_SUCCESS IIRC, otherwise execution fo UiApp will fails due to UNSUPPORTED return code. Another option is to simply pass  `-D DASHARO_SYSTEM_FEATURES_ENABLE=TRUE` to the edk2 build string

**Comment by pietrushnic** [comment]
@miczyg1 I have to use debug version because I have to provide you debug logs related with not reliable Secure Boot enabling/disabling (I'm pretty sure this is related with all variables storing). I also see that USB detection is not reliable.

Why we don't have `-D DASHARO_SYSTEM_FEATURES_ENABLE=TRUE`  as default in common-base-rebased?

**Comment by pietrushnic** [comment]
Ok, I see it can be passed in following way: `CONFIG_EDK2_CUSTOM_BUILD_PARAMS="-D DASHARO_SYSTEM_FEATURES_ENABLE=TRUE"`

**Pull request closed without merging**